### PR TITLE
Allow statinamic to be installed from Github

### LINKED ISSUE #203: Consume statinamic directly from Github
For trying new feature ... without the symlink trick

### PR DESCRIPTION
**Pull request by thangngoc89**
Close #203 

Some gotchas:

- Doesn't work with npm > 3.3.2 (see https://github.com/npm/npm/issues/10686)
- Need to rewrite cli setup command

comander can't resolve `statinamic-setup` since the `process.cwd()` is at `scripts/` and not `lib/`. so no commander sub-command

poke @MoOx can you rewrite this part when you're fixing #262 ?

### CODE REVIEW

**Comment by thangngoc89** [comment]
CI build cannot catch this above bug with `statinamic setup` command

**Comment by MoOx** [comment]
Awesome. Will take the relay from here ;)

**Comment by thangngoc89** [comment]
@MoOx PR updated. It's working now.

File: npm/postinstall.js [hunk 1]
```diff
@@ -27,6 +27,6 @@ stat("lib", function(error, stats1) {
   installer.on("error", function(err) {
     console.error(`Failed to build ${ pkg.name } automatically. `)
 
-    throw err
+    console.error(err)
```
Comment (MoOx) [inline npm/postinstall.js]: Why this change?

Comment (thangngoc89) [inline npm/postinstall.js]: ouch. I forgot to remove this commit why rebasing this branch. I made this change for debugging only.

Comment (thangngoc89) [inline npm/postinstall.js]: *while

File: npm/postinstall.js [hunk 2]
```diff
@@ -13,10 +13,16 @@ stat("lib", function(error, stats1) {
...
     "Builded sources not found. It looks like you might be attempting " +
     `to install ${ pkg.name } from git. ` +
     "Sources need to be transpiled before use and this will require you to " +
-    `have babel-cli installed as well as ${ pkg.babel.presets }.`
+    `have babel-cli installed as well as ${ pkg.babel.presets }.\n` +
+    "-".repeat(40) + "\n" +
+    "TL;DR;\n" +
+    "Type this command\n" +
+    "npm install babel-core babel-cli " + pkg.babel.presets.join(" ") +
+    " && npm rebuild statinamic"
```
Comment (MoOx) [inline npm/postinstall.js]: I will try to make this automatic.

**Comment by MoOx** [comment]
Now it works with `npm install https://github.com/MoOx/statinamic#github-install && statinamic setup`.

**Comment by MoOx** [comment]
I merged this too quickly, it's not working on windows :/

**Comment by thangngoc89** [comment]
What is the error with Windows?

**Comment by MoOx** [comment]
https://ci.appveyor.com/project/MoOx/statinamic/build/1003#L46

postinstall is ran on CI, I will just skip this for now.

**Comment by thangngoc89** [comment]
Most of answer on SO pointed to this repo.
https://github.com/IndigoUnited/node-cross-spawn

A drop in replacement for spawn in Windows

File: npm/postinstall.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    "Sources need to be transpiled before use. This may take a moment." +
+    "\n"
+  )
+
+  const spawnOpts = {
+    stdio: "inherit",
+    cwd: join(__dirname, "../"),
+  }
+
+  const fail  = (err) => {
```
Comment (thangngoc89) [inline npm/postinstall.js]: I'll try to avoid arrow function here.

File: npm/postinstall.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
+const stat = require("fs").stat
```
Comment (thangngoc89) [inline npm/postinstall.js]: I don't think you should use const here. node 4 (actually older V8 version) required "use strict" mode when using const.

File: npm/postinstall.js (hunk 3 above)
Comment (MoOx) [inline npm/postinstall.js]: this is working on node 4/5

File: npm/postinstall.js (hunk 4 above)
Comment (MoOx) [inline npm/postinstall.js]: it's working isn't it? If not this will raise a syntax error.

File: npm/postinstall.js (hunk 3 above)
Comment (thangngoc89) [inline npm/postinstall.js]: Haha. FYI https://github.com/MoOx/statinamic/issues/299#issuecomment-197163115

File: npm/postinstall.js (hunk 4 above)
Comment (thangngoc89) [inline npm/postinstall.js]: Well, I can't make this to work on node 5 so can't comment.

File: npm/postinstall.js (hunk 3 above)
Comment (MoOx) [inline npm/postinstall.js]: yeah. This is already "visible" per travis and appveyor yml.

Comment (thangngoc89) [inline npm/postinstall.js]: The user won't notice that. The report a bug immedialy when they saw it. (which is related to node version).
